### PR TITLE
Add test start, end date and test station to the next gen UI

### PR DESCRIPTION
**Pull request by martonmiklos**
Following upon this:
https://github.com/inventree/InvenTree/pull/6149#issuecomment-2002637842

Here is a PR which adds the new test fields to the React UI.

### CODE REVIEW

**Comment by martonmiklos** [comment]
> While you are likely still working on this:
> 

Dauh, my bad. I will fix it tomorrow...

File: src/frontend/src/components/forms/fields/ApiFormField.tsx [hunk 1]
```diff
@@ -216,6 +217,8 @@ export function ApiFormField({
         );
       case 'date':
         return <DateField controller={controller} definition={definition} />;
+      case 'datetime':
+        return <DateField controller={controller} definition={definition} />;
```
Comment (wolflu05) [inline src/frontend/src/components/forms/fields/ApiFormField.tsx:221]: This could be simplified to using two case blocks with one return as both have the same arguments like done with int, float, ...

File: src/frontend/src/components/forms/fields/DateField.tsx [hunk 2]
```diff
@@ -11,20 +13,25 @@ export default function DateField({
   controller: UseControllerReturn<FieldValues, any>;
   definition: ApiFormFieldType;
 }) {
+  dayjs.extend(customParseFormat);
```
Comment (wolflu05) [inline src/frontend/src/components/forms/fields/DateField.tsx:16]: Wouldn't this make sense to move this outside of the component into the global scope of this file, as this function is called a lot of times (on every change of the value)?

**Comment by wolflu05** [comment]
BTW, it would be great if you could stop force pushing as this may break review comments and makes it really hard to follow along the changes. Just push new changes as new commits, they are squash merged anyway.

File: src/frontend/src/components/forms/fields/DateField.tsx (hunk 2 above)
Comment (martonmiklos) [reply src/frontend/src/components/forms/fields/DateField.tsx:16]: I am new to the PUI and the typescipt thing, where is exactly  the global scope of this file defined?

Comment (wolflu05) [reply src/frontend/src/components/forms/fields/DateField.tsx:16]: Just outside the function directly in the file after the imports. Because this line doesn't seem to be dependent on any react stuff.

Comment (martonmiklos) [reply src/frontend/src/components/forms/fields/DateField.tsx]: Ah thanks, I thought that it should be moved to some more global file.

**Comment by SchrodingersGat** [comment]
LGTM - @wolflu05 are you happy with this also?

**Review by wolflu05 — APPROVED** [review]
Yes, LGTM
